Explorer++/Helper/FileOperations.cpp

```diff
@@ -649,3 +671,77 @@ void FileOperations::DeleteFileSecurely(const std::wstring &strFilename,
 
 	DeleteFile(strFilename.c_str());
 }
+
+HRESULT FileOperations::Undelete(const PCIDLIST_ABSOLUTE &pidl)
+{
+	wil::com_ptr_nothrow<IShellFolder> pDesktop{};
+	HRESULT hr = SHGetDesktopFolder(&pDesktop);
+	if (FAILED(hr) || !pDesktop)
+		return hr;
+
+	CComHeapPtr<ITEMIDLIST_ABSOLUTE> pidlbin{};
+	hr = SHGetKnownFolderIDList(FOLDERID_RecycleBinFolder, KF_FLAG_DEFAULT, NULL, &pidlbin);
+	if (FAILED(hr) || !pidlbin)
+		return hr;
+
+	wil::com_ptr_nothrow<IShellFolder> pShellFolder{};
+	hr = pDesktop->BindToObject(pidlbin, nullptr, IID_PPV_ARGS(&pShellFolder));
+	if (FAILED(hr) || !pShellFolder)
+		return hr;
+
+	wil::com_ptr_nothrow<IEnumIDList> pEnum{};
+	hr = pShellFolder->EnumObjects(NULL, SHCONTF_FOLDERS | SHCONTF_NONFOLDERS, &pEnum);
+	if (FAILED(hr) || !pEnum)
+		return hr;
+
+	ULONG fetched{};
+	for (PITEMID_CHILD pidChild{}; pEnum->Next(1, &pidChild, &fetched) == S_OK; pidChild = nullptr)

```

Comment:
`unique_pidl_child` should be used when enumerating to ensure the pidl is freed. See, for example, https://github.com/derceg/explorerplusplus/blob/47bcbd41e87826c47ee201ab1c85820439cb9028/Explorer%2B%2B/Explorer%2B%2B/ShellEnumerator.cpp#L29-L31.